session_test.go

```diff
@@ -168,6 +168,27 @@ func (s *S) TestURLInvalidReadPreference(c *C) {
 	}
 }
 
+func (s *S) TestURLUnixSocket(c *C) {
+	type test struct {
+		url      string
+		socket   string
+		database string
+	}
+
+	tests := []test{
```

Comment:
Thanks for including tests, it makes maintenance in the future so much easier for everyone 👍